tests/unit/nupic/encoders/random_distributed_scalar_test.py

```diff
@@ -428,5 +436,40 @@ def testEncodeInvalidInputType(self):
 
 
 
+  def testCapNProtoSerialization(self):
+    original = RandomDistributedScalarEncoder(name='enc', resolution=1.0, w=23,
+                                              n=500, offset = 0.0)
+
+    originalValue = original.encode(1)
+
+    proto1 = RandomDistributedScalarEncoderProto.new_message()
+    original.write(proto1)
+
+    # Write the proto to a temp file and read it back into a new proto
+    with tempfile.TemporaryFile() as f:
+      proto1.write(f)
+      f.seek(0)
+      proto2 = RandomDistributedScalarEncoderProto.read(f)
+
+    encoder = RandomDistributedScalarEncoder.read(proto2)
+
+    self.assertIsInstance(encoder, RandomDistributedScalarEncoder)
+    self.assertEqual(encoder.resolution, original.resolution)
+    self.assertEqual(encoder.w, original.w)
+    self.assertEqual(encoder.n, original.n)
+    self.assertEqual(encoder.name, original.name)
+    self.assertEqual(encoder.verbosity, original.verbosity)
+    self.assertEqual(encoder.minIndex, original.minIndex)
+    self.assertEqual(encoder.maxIndex, original.maxIndex)
+    self.assertTrue(numpy.array_equal(encoder.encode(1), originalValue))
+    self.assertEqual(original.decode(encoder.encode(1)),
+                     encoder.decode(original.encode(1)))
+    self.assertEqual(original.random.getSeed(), encoder.random.getSeed())
+
+    for key, value in original.bucketMap.items():
+      self.assertTrue(numpy.array_equal(value, encoder.bucketMap[key]))
+
```

Comment:
How is that different from https://github.com/numenta/nupic/pull/1972#discussion-diff-27497574R464 ?